File: 100 Days Code/DAY 10/Rotate_image.cpp
//! Rotate Image

#include <bits/stdc++.h>
using namespace std;
void rotate(vector<vector<int>> &matrix) {
  int n = matrix.size();
  int m = matrix[0].size();
  // Transpose Of Matrix
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < i; j++) {
      swap(matrix[i][j], matrix[j][i]);
    }
  }
  // Row-wise reverse
  for (int i = 0; i < n; i++) {
    reverse(matrix[i].begin(), matrix[i].end());
  }
}
int main() {
  int n, m;
  cin >> n >> m;
  vector<vector<int>> arr(n, vector<int>(m));
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < m; j++) {
      cin >> arr[i][j];
    }
  }
  rotate(arr);
  for (int i = 0; i < n; i++) {
    for (int j = 0; j < m; j++) {
      cout << arr[i][j] << " ";
    }
    cout << endl;
  }
  return 0;
}

File: JavaChallenge/Day5/Rotate_Image.java

class Solution {
    public void rotate(int[][] arr) {
        int m = arr.length;
        int n = arr[0].length;
        // transpose of matrix
           for (int i = 0; i < m; i++) {
            for (int j = 0; j < i; j++) {  
                int temp = arr[i][j];
                arr[i][j] = arr[j][i];
                arr[j][i] = temp;
            }            
        }
        // for reversing the row of transposed matrix 
       
        for(int i = 0 ; i<m;i++){
             int a =0 ; 
             int b=m-1;
            while(a<b){
                int temp = arr[i][a];
                arr[i][a]= arr[i][b];
                arr[i][b]= temp;
                a++;
                b--;
            }
        }
    }     
}
